dev/src/write-batch.ts

```diff
@@ -509,6 +543,51 @@ export class WriteBatch {
     return this.commit_();
   }
 
+  /**
+   * Commits all pending operations to the database and verifies all
+   * preconditions. Fails the entire write if any precondition is not met.
+   *
+   * The writes in the batch are not applied atomically and can be applied out
+   * of order.
+   *
+   * @private
+   */
+  async batchCommit(commitOptions?: {
+    requestTag?: string;
+  }): Promise<BatchWriteResult[]> {
+    this._committed = true;
+    const tag = (commitOptions && commitOptions.requestTag) || requestTag();
+    await this._firestore.initializeIfNeeded(tag);
+
+    const database = this._firestore.formattedName;
+    const request: api.IBatchWriteRequest = {database};
+    const writes = this._ops.map(op => op());
+    request.writes = [];
+
+    for (const req of writes) {
+      if (req.precondition) {
+        req.write!.currentDocument = req.precondition;
+      }
+      request.writes.push(req.write);
+    }
+
+    const response = await this._firestore.request<
+      api.IBatchWriteRequest,
+      api.BatchWriteResponse
+    >('batchWrite', request, tag);
+
+    if (request.writes!.length > 0) {
```

Comment:
You can drop this if check. map() will do the right thing.

Here and below.

Reply:
good point, dropped